File: mobile/modules/native-menu/android/src/main/java/expo/modules/nativemenu/NativeMenuModule.kt
package expo.modules.nativemenu

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeMenuModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NativeMenu")

    Constants {
      mapOf("isNativeMenuAvailable" to true)
    }

    View(NativeMenuView::class) {
      Events("onSelect", "onPress")

      Prop("itemsJson") { view: NativeMenuView, json: String ->
        view.itemsJson = json
      }

      Prop("trigger") { view: NativeMenuView, trigger: String ->
        view.trigger = trigger
      }

      Prop("accessibilityLabel") { view: NativeMenuView, label: String? ->
        view.contentDescription = label
      }

      Prop("testID") { view: NativeMenuView, testID: String? ->
        view.tag = testID
      }

      Prop("disabled") { view: NativeMenuView, disabled: Boolean ->
        view.isMenuDisabled = disabled
      }
    }
  }
}

File: mobile/modules/native-menu/ios/NativeMenuModule.swift
import ExpoModulesCore

/// Bridge for a system menu attached to a view. iOS uses `UIButton.menu` / `UIMenu`.
/// Android’s sibling module presents an anchored `PopupMenu` on the same view.
public class NativeMenuModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NativeMenu")

    Constants([
      "isNativeMenuAvailable": true
    ])

    View(NativeMenuView.self) {
      Events("onSelect", "onPress")

      Prop("itemsJson") { (view: NativeMenuView, json: String) in
        view.itemsJson = json
      }

      Prop("trigger") { (view: NativeMenuView, trigger: String) in
        view.trigger = trigger
      }

      Prop("accessibilityLabel") { (view: NativeMenuView, label: String?) in
        view.accessibilityTitle = label
      }

      Prop("testID") { (view: NativeMenuView, testID: String?) in
        view.testIdentifier = testID
      }

      Prop("disabled") { (view: NativeMenuView, disabled: Bool) in
        view.isMenuDisabled = disabled
      }
    }
  }
}
